sql-plugin-api/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala

```diff
@@ -310,6 +310,9 @@ object ShimLoader {
     // hack for databricks, try to find something more reliable?
     if (SPARK_BUILD_USER.equals("Databricks")) {
       SPARK_VERSION + "-databricks"
+    // hack for preview versions to act as the base version
+    } else if (SPARK_VERSION matches """\d+\.\d+\.\d+-preview\d+""") {
+      SPARK_VERSION.split('-')(0)
```

Comment:
Is this for is_spark_400_or_later ?

Reply:
What aspect of it?

Preview versions of spark are not supported by us, but this change is necessary for us to test preview versions of spark as if they are the actual version. The version numbering system holds true for Spark 3, but they have been inconsistent about numbering the preview version (I think partly because they're not sure how many previews they will need).

Reply:
I think we should just change the 400 provider to accept SNAPSHOT and the latest `preview2`, dropping preview1 https://github.com/NVIDIA/spark-rapids/blob/a41a6160ee3783a9cffd623b46e00992c1204eff/sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/spark400/SparkShimServiceProvider.scala#L26